src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseMessage.cs

```diff
@@ -181,19 +183,62 @@ private static byte[] DecodeSignature(CborReader reader)
             return reader.ReadByteString();
         }
 
-        internal static int CreateToBeSigned(string context, ReadOnlySpan<byte> encodedProtectedHeader, ReadOnlySpan<byte> content, Span<byte> destination)
+        internal static int CreateHashedToBeSigned(Span<byte> destination, string context, ReadOnlySpan<byte> encodedProtectedHeader, ReadOnlySpan<byte> content, HashAlgorithmName hashAlgorithm)
+        {
+            int bytesWritten = CreateToBeSigned(destination, context, encodedProtectedHeader, ReadOnlySpan<byte>.Empty);
+            bytesWritten -= 1; // Trim the empty bstr content, it is just a placeholder.
+
+            using (var hasher = IncrementalHash.CreateHash(hashAlgorithm))
+            {
+                hasher.AppendData(destination.Slice(0, bytesWritten));
+
+                // content length
+                CoseHelpers.WriteByteStringLength(hasher, (ulong)content.Length);
+
+                //content
+                hasher.AppendData(content);
+
+                return hasher.GetHashAndReset(destination);
+            }
+        }
+
+        internal static async Task<int> CreateHashedToBeSignedAsync(byte[] destination, string context, ReadOnlyMemory<byte> encodedProtectedHeader, Stream content, HashAlgorithmName hashAlgorithm)
+        {
+            int bytesWritten = CreateToBeSigned(destination, context, encodedProtectedHeader.Span, ReadOnlySpan<byte>.Empty);
+            bytesWritten -= 1; // Trim the empty bstr content, it is just a placeholder.
+
+            using (IncrementalHash hasher = IncrementalHash.CreateHash(hashAlgorithm))
+            {
+                hasher.AppendData(destination, 0, bytesWritten);
+
+                //content length
+                CoseHelpers.WriteByteStringLength(hasher, (ulong)(content.Length - content.Position));
+
+                // content
+                byte[] buffer = new byte[4096];
+                int bytesRead;
+#pragma warning disable CA1835 // Prefer the 'Memory'-based overloads for 'ReadAsync' and 'WriteAsync'
```

Comment:
It's a little nicer on the runtime if instead of suppressing this you just do a `#if NETFRAMEWORK` and call the newer ones in newer runtimes.